Modules/StudyProgramme/classes/class.ilStudyProgrammeUserProgressDB.php

```diff
@@ -0,0 +1,173 @@
+	}
+
+	/**
+	 * Get the instance for an assignment.
+	 *
+	 * Throws when the node does not belong to the assignment.
+	 *
+	 * @throws ilException
+	 * @param  int $a_program_id
+	 * @param  int $a_user_id
+	 * @return ilStudyProgrammeUserProgress
+	 */
+	static public function getInstancesForAssignment($a_assignment_id) {
+		$progresses = ilStudyProgrammeProgress::where(array
+							( "assignment_id" => $a_assignment_id
+							))->get();
+		if (count($progresses) == 0) {
+			require_once("Modules/StudyProgramme/classes/exceptions/class.ilStudyProgrammeNoProgressForAssignmentException.php");
+			throw new ilStudyProgrammeNoProgressForAssignmentException
+								("ilStudyProgrammeUserProgress::getInstancesForAssignment: "
+								."Can't find progresses for assignment '$a_assignment_id'.");
+		}
+		return array_map(function($dat) {
+			return new ilStudyProgrammeUserProgress($dat);
+		}, $progresses);
+	}
+
+	/**
+	 * Get the instances for a program node.
+	 *
+	 * @param int $a_program_id
+	 * @return ilStudyProgrammeUserProgress[]
+	 */
+	static public function getInstancesForProgram($a_program_id) {
+		$progresses = ilStudyProgrammeProgress::where(array
+							( "prg_id" => $a_program_id
+							))->get();
+		return array_values(array_map(function($dat) {
+			return new ilStudyProgrammeUserProgress($dat);
+		}, $progresses));
+	}
+
+	/**
+	 * Get a user readable representation of a status.
+	 */
+	public function statusToRepr($a_status) {
+		global $DIC;
+		$lng = $DIC['lng'];
+		$lng->loadLanguageModule("prg");
+
+		if ($a_status == ilStudyProgrammeProgress::STATUS_IN_PROGRESS) {
+			return $lng->txt("prg_status_in_progress");
+		}
+		if ($a_status == ilStudyProgrammeProgress::STATUS_COMPLETED) {
+			return $lng->txt("prg_status_completed");
+		}
+		if ($a_status == ilStudyProgrammeProgress::STATUS_ACCREDITED) {
+			return $lng->txt("prg_status_accredited");
+		}
+		if ($a_status == ilStudyProgrammeProgress::STATUS_NOT_RELEVANT) {
+			return $lng->txt("prg_status_not_relevant");
+		}
+		throw new ilException("Unknown status: '$a_status'");
+	}
+
+
+
+	const ACTION_MARK_ACCREDITED = "mark_accredited";
+	const ACTION_UNMARK_ACCREDITED = "unmark_accredited";
+	const ACTION_SHOW_INDIVIDUAL_PLAN = "show_individual_plan";
+	const ACTION_REMOVE_USER = "remove_user";
+	/**
+	 * Get a list with possible actions on a progress record.
+	 *
+	 * @param int $a_node_id	object_id!
+	 * @param int $a_belongs	object_id!
+	 * @param int $a_status
+	 */
+	public function getPossibleActions($a_node_id, $a_root_prg_id, $a_status) {
+		$actions = array();
+		if ($a_node_id == $a_root_prg_id) {
+			$actions[] = self::ACTION_SHOW_INDIVIDUAL_PLAN;
+			$actions[] = self::ACTION_REMOVE_USER;
+		}
+		if ($a_status == ilStudyProgrammeProgress::STATUS_ACCREDITED) {
+			$actions[] = self::ACTION_UNMARK_ACCREDITED;
+		}
+		else if ($a_status == ilStudyProgrammeProgress::STATUS_IN_PROGRESS) {
+			$actions[] = self::ACTION_MARK_ACCREDITED;
+		}
+		return $actions;
+	}
```

Comment:
I know I told you to move all statics to this new class, but this most probably does not belong to a `DB`-class. Do we ever need that without a concrete instance of `ilStudyProgrammeUserProgress`? If not, this could as well become a non-static method on the class.